docs/docs/plugin-authoring.md

```diff
@@ -0,0 +1,61 @@
+---
+title: Plugin Authoring
+---
+
+Creating custom plugins in Gatsby is a breeze with our straightforward authoring process.
+
+## Core Concepts
+
+- Each Gatsby plugin can be installed as an npm package or as a [local plugin](#local-plugins)
+- At minimum, a `package.json` is required
+- A plugin has access to the the Gatsby [Node](/docs/node-apis/), [SSR](/docs/ssr-apis/), and [browser](/docs/browser-apis/) APIs
+
+## Plugin naming conventions
+
+There are four standard plugin naming conventions for Gatsby:
+
+- **`gatsby-source-*`** — a source plugin loads data from a given source (e.g. WordPress, MongoDB, the file system). Use this plugin type if you are connecting a new source of data to Gatsby.
+  - Example: `gatsby-source-contentful`
+  - Docs: [create a source plugin](/docs/create-source-plugin/)
+- **`gatsby-transformer-*`** — a transformer plugin converts data from one format to another (e.g. CSV to JSON). Use this naming convention
```

Comment:
"A transformer plugin converts data from one format into JavaScript objects"

I can't think of any reason why a transformer plugin wouldn't ever transform data directly into JavaScript objects. It *could* happen but it's unlikely.